src/cmd/cli/command/compose.go

```diff
@@ -37,6 +37,20 @@ func GetUnreferencedManagedResources(serviceInfos []*defangv1.ServiceInfo) []str
 	return managedResources
 }
 
+func startTimer(ctx context.Context, waitTimeoutSec int, onTimeoutHandler func(error)) {
+	if waitTimeoutSec >= 0 { // Checks if wait-timeout value is valid
+		timeoutCtx, timeoutCancel := context.WithTimeout(ctx, time.Duration(waitTimeoutSec)*time.Second)
```

Comment:
`context.WithTimeout` returns a context that gets cancelled, so that should be all we need to pass to `cli.WaitServiceState` (or even `cli.Tail`). No need for us to have our own Go-routine.

Reply:
I've made the change so it gets passed to `cli.WaitServiceState`. Also, I made some additional changes to `tail.go` and `commands.go` in order to suppress some errors that show up but are irrelevant to the user, because they are the one who requested the timeout to occur.